src/clj_kondo/impl/linters.clj

```diff
@@ -121,6 +121,21 @@
       ;; (prn "tags" tags)
       (types/lint-arg-types ctx called-fn arg-types tags call))))
 
+(defn lint-bangs!
+  [ctx filename fn-name in-def]
+  (when in-def
+    (let [has-bang? (-> fn-name str (str/ends-with? "!"))]
```

Comment:
Out of curiosity, was the lack of when-let a choice here?

Reply:
No, just a glitch.